spring-amqp/src/main/java/org/springframework/amqp/core/AbstractBuilder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.core;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Base class for builders supporting arguments.
+ *
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+public abstract class AbstractBuilder {
```

Comment:
I thought about that but a) the MessageBuilder is here and b) it will cause a package tangle (template references support) - and support will reference core - we generally don't have support classes reference other packages.